src/mongoose_http_client.erl

```diff
@@ -151,20 +134,17 @@ make_request(Pool, Path, Method, Headers, Query) ->
             {error, {'EXIT', Reason}};
         {error, Reason} ->
             {error, Reason}
+    catch
+        exit:timeout ->
+            {error, pool_timeout};
+        exit:no_workers ->
+            {error, pool_down};
+        Type:Error ->
+            {error, {Type, Error}}
     end.
 
-start_supervisor() ->
-    Proc = sup_proc_name(),
-    ChildSpec =
-        {Proc,
-         {Proc, start_link, [Proc]},
-         permanent,
-         infinity,
-         supervisor,
-         [Proc]},
-    ejabberd_sup:start_child(ChildSpec).
-
-stop_supervisor() ->
-    Proc = sup_proc_name(),
-    ejabberd_sup:stop_child(Proc).
+pool_name(PoolName) ->
+    list_to_existing_atom("mongoose_http_client_pool_" ++ atom_to_list(PoolName)).
 
+pool_name(new, PoolName) ->
```

Comment:
Do you think naming this function `make_pool_name` would more informative? In my opinion it would be. Reading just `pool_name(new, PoolName)` I don't know if this will combine the first arg with the second or what, while reading `make_pool_name(PoolName)` is more self-descriptive.